net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java

```diff
@@ -0,0 +1,84 @@
+package net.sf.eclipsecs.core.config.meta;
+
+import java.util.Set;
+
+import com.puppycrawl.tools.checkstyle.PackageNamesLoader;
+import com.puppycrawl.tools.checkstyle.PackageObjectFactory;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import net.sf.eclipsecs.core.CheckstylePlugin;
+
+public final class CheckUtil {
+    private CheckUtil() {
+    }
+
+    public static Object getCheck(String checkName) {
+        final ClassLoader classLoader = CheckstylePlugin.getDefault()
+                .getAddonExtensionClassLoader();
+        try {
+            final Set<String> packageNames = PackageNamesLoader.getPackageNames(classLoader);
+            return new PackageObjectFactory(packageNames, classLoader)
+                    .createModule(checkName);
+        }
+        catch (CheckstyleException ex) {
+            throw new IllegalStateException("exception occured during load of " + checkName, ex);
+        }
+    }
+
+    public static String getAcceptableTokens(String checkName) {
+        final Object checkResult = getCheck(checkName);
+        String result = null;
+        if (AbstractJavadocCheck.class.isAssignableFrom(checkResult.getClass())) {
+            final AbstractJavadocCheck javadocCheck = (AbstractJavadocCheck) checkResult;
+            result = getTokenText(true, javadocCheck.getAcceptableJavadocTokens(),
+                    javadocCheck.getRequiredJavadocTokens());
+        }
+        else if (AbstractCheck.class.isAssignableFrom(checkResult.getClass())) {
+            final AbstractCheck check = (AbstractCheck) checkResult;
+            result = getTokenText(false, check.getAcceptableTokens(),
+                    check.getRequiredTokens());
+        }
+        return result;
+    }
+
+    public static String getTokenText(boolean isJavadocCheck, int[] tokens, int... requiredTokens) {
```

Comment:
please make method to be like `getTokenText(tokens, JavadocUtil::getTokenName)` 
filtering out `requiredTokens` should be in separate method.
this will keep Utils to be simple, context less and very reusable.

Reply:
first update to avoid boolean.
```
$ git diff net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java
diff --git a/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java b/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java
index fea5a0d..a97b9ba 100644
--- a/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java
+++ b/net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java
@@ -1,6 +1,7 @@
 package net.sf.eclipsecs.core.config.meta;
 
 import java.util.Set;
+import java.util.function.Function;
 
 import com.puppycrawl.tools.checkstyle.PackageNamesLoader;
 import com.puppycrawl.tools.checkstyle.PackageObjectFactory;
@@ -21,12 +22,12 @@ public final class CheckUtil {
       String result = null;
       if (AbstractJavadocCheck.class.isAssignableFrom(checkResult.getClass())) {
           final AbstractJavadocCheck javadocCheck = (AbstractJavadocCheck) checkResult;
-          result = getTokenText(true, javadocCheck.getAcceptableJavadocTokens(),
+          result = getTokenText(JavadocUtil::getTokenName, javadocCheck.getAcceptableJavadocTokens(),
                   javadocCheck.getRequiredJavadocTokens());
       }
       else if (AbstractCheck.class.isAssignableFrom(checkResult.getClass())) {
           final AbstractCheck check = (AbstractCheck) checkResult;
-          result = getTokenText(false, check.getAcceptableTokens(),
+          result = getTokenText(JTokenUtil::getTokenName, check.getAcceptableTokens(),
                   check.getRequiredTokens());
       }
       else {
@@ -49,7 +50,7 @@ public final class CheckUtil {
         }
     }
 
-    private static String getTokenText(boolean isJavadocCheck, int[] tokens, int... requiredTokens) {
+    private static String getTokenText(Function<Integer, String> function, int[] tokens, int... requiredTokens) {
         final StringBuilder result = new StringBuilder();
         boolean first = true;
 
@@ -74,12 +75,7 @@ public final class CheckUtil {
                 result.append(',');
             }
 
-            if (isJavadocCheck) {
-                result.append(JavadocUtil.getTokenName(token));
-            }
-            else {
-                result.append(TokenUtil.getTokenName(token));
-            }
+            result.append(function.apply(token));
         }
 
         return result.toString();
```

Reply:
after update above, please do code to be like (imaginary code):
```
          int[] modifiableTokens = substract(check.getAcceptableTokens(), check.getRequiredTokens());
          result = getTokenText(TokenUtil::getTokenName, modifiableTokens);
```

Reply:
Done

Reply:
good, but please do like:
`result = getTokenText(TokenUtil::getTokenName, modifiableTokens);`
nested methods calls are hard to read.

Reply:
Done